src/core/shape_inference/include/broadcast_shape_inference.hpp

```diff
@@ -49,118 +49,112 @@ void validate_target_shape_none(const ov::Node* op,
             if (arg_shape.size() > 0) {
                 NODE_VALIDATION_CHECK(
                     op,
-                    target_shape[axes_mapping_val[i]].compatible(arg_shape[i]) || arg_shape[i].compatible(1),
+                    target_input_shape[axes_mapping_val[i]].compatible(arg_shape[i]) || arg_shape[i].compatible(1),
                     "Broadcast target[axes_mapping[",
                     i,
                     "]]",
                     " Expected ",
                     arg_shape[i],
                     ". Got ",
-                    target_shape[axes_mapping_val[i]]);
+                    target_input_shape[axes_mapping_val[i]]);
             }
         }
     }
 }
 
 template <typename T>
-void validate_target_shape_numpy(const ov::Node* op, const T& arg_shape, const T& target_shape) {
-    if (arg_shape.rank().is_dynamic() || target_shape.rank().is_dynamic()) {
+void validate_target_shape_numpy(const ov::Node* op, const T& arg_shape, const T& target_input_shape) {
+    if (arg_shape.rank().is_dynamic() || target_input_shape.rank().is_dynamic()) {
         return;
     }
     const auto arg_rank_length = arg_shape.size();
-    const auto target_rank_length = target_shape.size();
-    const auto start_axis = target_rank_length - arg_rank_length;
+    const auto target_rank_length = target_input_shape.size();
     NODE_VALIDATION_CHECK(op,
-                          start_axis >= 0,
```

Comment:
The result of `target_rank_length - arg_rank_length` was casted by auto to `size_t` type and the check to avoid smaller rank (`start_axis >= 0`) was always true. Fixed to avoid narrow casting.